src/bin/server.ts

```diff
@@ -46,16 +46,67 @@ interface Args {
   if (configPath) {
     try {
       options = require(configPath);
-      // make sure that networkKey is passed as buffer.
-      // accept both zwave2mqtt format as ozw format
-      if (options.networkKey && options.networkKey.length === 32) {
-        options.networkKey = Buffer.from(options.networkKey, "hex");
+      const securityKeyNames = [
+        "S0_Legacy",
+        "S2_AccessControl",
+        "S2_Authenticated",
+        "S2_Unauthenticated",
+      ];
+      // We prefer the securityKeys option over the networkKey one
+      for (const key of securityKeyNames) {
+        if (
+          options.securityKeys &&
+          options.securityKeys[key] &&
+          options.securityKeys[key].length === 32
+        ) {
+          options.securityKeys[key] = Buffer.from(
+            options.securityKeys[key],
+            "hex"
+          );
+        } else if (
+          options.securityKeys &&
+          options.securityKeys[key] &&
+          options.securityKeys[key].includes("0x")
+        ) {
+          options.securityKeys[key] = options.securityKeys[key]
+            .replace(/0x/g, "")
+            .replace(/, /g, "");
+          options.securityKeys[key] = Buffer.from(
+            options.securityKeys[key],
+            "hex"
+          );
+        } else if (options.securityKeys && options.securityKeys[key]) {
+          console.error(
+            "Error: Invalid securityKeys." + key + " value defined."
+          );
+          return;
+        }
+      }
+      // If both securityKeys.S0_Legacy and networkKey are defined, prefer securityKeys.S0_Legacy
+      if (options.securityKeys.S0_Legacy && options.networkKey) {
+        console.warn(
+          "`networkKey` option is being ignored because `securityKeys.S0_Legacy` option is present"
```

Comment:
When it comes to setting security keys, let's not "ignore" things but raise errors.

Reply:
I would prefer an error also. You don't want to let users do something like define different keys for `S0_Legacy` and `networkKey`. The `S0_Legacy` key needs to be the old key to preserve security capabilities for already included devices. node-zwave-js will prefer `S0_Legacy` if both are provided, and it will use `networkKey` if `S0_Legacy` is not provided.

If not an error, than I would say it's preferable to choose `networkKey` instead because it would have been set in a previous version and `S0_Legacy` might be a new key created in confusion.

Reply:
now we throw an error